File: ObjectOrientedProgramming/08. EncapsulationAndPolymorphism/01. Shapes/Circle.cs
//•	Define a class Circle with a suitable constructor and IShape.

namespace _01.Shapes
{
    using System;

    public class Circle : IShape
    {
        private double radius;

        public Circle(double radius)
        {
            this.Radius = radius;
        }

        public double Radius { get; set; }

        public double CalculateArea()
        {
            double area = Math.PI * this.Radius * this.Radius;

            return area;
        }

        public double CalculatePerimeter()
        {
            double perimeter = 2 * Math.PI * this.Radius;

            return perimeter;
        }
    }
}

File: ObjectOrientedProgramming/10. OOPInJava/src/pkg10/oopinjava/geometry/Circle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package pkg10.oopinjava.geometry;

/**
 *
 * @author Daniel Todorov
 */
public class Circle extends PlaneShape {

    private double radius;

    public Circle(Vertix2D vertix, double radius) {
        super(new Vertix2D[]{vertix});
        
        this.setRadius(radius);
    }

    /**
     * @return the radius
     */
    public final double getRadius() {
        return radius;
    }

    /**
     * @param radius the radius to set
     */
    public final void setRadius(double radius) {
        this.radius = radius;
    }

    @Override
    public double getPerimeter() {
        double perimeter = 2 * Math.PI * this.getRadius();
        
        return perimeter;
    }

    @Override
    public double getArea() {
        double area = Math.PI * this.getRadius() * this.getRadius();
        
        return area;
    }
}
